shared/lib_pv_io_manager.cpp

```diff
@@ -123,26 +124,32 @@ Irradiance_IO::Irradiance_IO(compute_module* cm, std::string cmName)
 
 		weatherDataProvider->rewind();
 	}
-	else if (weatherDataProvider->nrecords() == 8760)
+	else if (weatherDataProvider->annualSimulation() && weatherDataProvider->nrecords() == 8760)
 	{
 		// hourly file with no minute data column.  assume
 		// integrated/averaged values and use mid point convention for interpreting results
 		instantaneous = false;
 		tsShiftHours = 0.5;
 	}
 	else
-		throw exec_error(cmName, "subhourly weather files must specify the minute for each record");
+		throw exec_error(cmName, "subhourly and non-annual weather files must specify the minute for each record");
 
 	weatherDataProvider->header(&weatherHeader);
 
 	//total number of records in the weather file (i.e. 8760 * timestep)
 	numberOfWeatherFileRecords = weatherDataProvider->nrecords();
-	stepsPerHour = numberOfWeatherFileRecords / 8760;
-	dtHour = 1.0 / stepsPerHour;
+	dtHour = 1.0; //initialize these values to 1 for non-annual simulations
+	stepsPerHour = 1.0;
```

Comment:
Yes, I think this is going to be a tricky point for folks to follow. A PySAM example is a great idea. As far as I tried to program, everything uses the _timestamp_ correctly, but those things which would require the _time interval_ will not be done correctly, which we decided to do rather than calculate it from timestamps since folks might be entering discontinuous data.

I tried to include warnings or disallow cases where the time interval would be used incorrectly:
-batteries (not allowed with non-annual simulations)
-timestep beam shading calculations (not allowed)
-snow model (warning issued)
-wind stow model (pvwattsv7 only, warning issued)

I just realized I should add a warning that the transient thermal model in pvsamv1 won't be triggered, so I will take care of that. Are there any other pieces you can see that I'm missing? I searched for `dtHour`, `ts_hour`, and `steps_per_hour `in pvsamv1 to find all cases.

Reply:
I don't recall seeing any other model. I'm definitely agree that all the warnings are essential. My suggestion was just to have that information more explicit for people when they are getting into the model to know what its limitations are-- even the list written here would be good for future reference. :)